core/src/macosX64Main/kotlin/PeripheralDelegate.kt

```diff
@@ -0,0 +1,240 @@
+
+    sealed class Response {
+
+        abstract val peripheralIdentifier: NSUUID
+        abstract val error: NSError?
+
+        data class DidDiscoverServices(
+            override val peripheralIdentifier: NSUUID,
+            override val error: NSError?,
+        ) : Response()
+
+        data class DidDiscoverCharacteristicsForService(
+            override val peripheralIdentifier: NSUUID,
+            val service: CBService,
+            override val error: NSError?,
+        ) : Response()
+
+        data class DidWriteValueForCharacteristic(
+            override val peripheralIdentifier: NSUUID,
+            val characteristic: CBCharacteristic,
+            override val error: NSError?,
+        ) : Response()
+
+        data class DidUpdateValueForDescriptor(
+            override val peripheralIdentifier: NSUUID,
+            val descriptor: CBDescriptor,
+            override val error: NSError?,
+        ) : Response()
+
+        data class DidUpdateNotificationStateForCharacteristic(
+            override val peripheralIdentifier: NSUUID,
+            val characteristic: CBCharacteristic,
+            override val error: NSError?,
+        ) : Response()
+
+        data class IsReadyToSendWriteWithoutResponse(
+            override val peripheralIdentifier: NSUUID,
+            override val error: NSError?,
+        ) : Response()
+
+        data class DidReadRssi(
+            override val peripheralIdentifier: NSUUID,
+            val rssi: NSNumber,
+            override val error: NSError?,
+        ) : Response()
+    }
+
+    private val _response = Channel<Response>(BUFFERED)
+    val response: ReceiveChannel<Response> = _response
+
+    sealed class DidUpdateValueForCharacteristic {
+
+        abstract val cbCharacteristic: CBCharacteristic
+
+        data class Data(
+            override val cbCharacteristic: CBCharacteristic,
+            val data: NSData,
+        ) : DidUpdateValueForCharacteristic()
+
+        data class Error(
+            override val cbCharacteristic: CBCharacteristic,
+            val error: NSError,
+        ) : DidUpdateValueForCharacteristic()
+    }
+
+    // todo: MutableSharedFlow
+    private val _characteristicChanges = BroadcastChannel<DidUpdateValueForCharacteristic>(BUFFERED)
+    val characteristicChanges: Flow<DidUpdateValueForCharacteristic>
+        // todo: Remove `get()` when `_characteristicChanges` is MutableSharedFlow.
+        get() = _characteristicChanges.openSubscription().consumeAsFlow()
+
+    /* Discovering Services */
+
+    override fun peripheral(
+        peripheral: CBPeripheral,
+        didDiscoverServices: NSError?,
+    ) {
+        println("<- PeripheralDelegate didDiscoverServices")
+        _response.sendBlocking(DidDiscoverServices(peripheral.identifier, didDiscoverServices))
+    }
+
+    // todo: func peripheral(CBPeripheral, didDiscoverIncludedServicesFor: CBService, error: Error?)
+
+    /* Discovering Characteristics and their Descriptors */
```

Comment:
When doing this style of comments, have you considered doing `//region Discovering Services` and `//endregion`?  I think they're a little uglier when just reading the code, but they enable folding in IntelliJ which can be worth it sometimes. 🤷  up to you.

Reply:
I definitely see the benefit, though I personally don't use folding at all — so, happy to have it be `region` but I slightly prefer it how it is just so it looks nicer. 🤷

Reply:
I'd leave it as is then. I know some people who love folding, but if the main dev on the project doesn't need it, no reason to clutter it up.